enumerable.rb

```diff
@@ -0,0 +1,199 @@
+
+    arr = check_data_type(self)
+    for i in arr
+      yield(i)
+    end
+    arr
+  end
+
+  ## my_each_index method
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = check_data_type(self)
+    i = 0
+    until i >= arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+    arr
+  end
+
+  ## my_select method
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each { |i| arr << i if yield(i) }
+    arr
+  end
+
+  ## my_any method
+  def my_any?
+    unless block_given?
+      for i in self
+        return true if i == true
+      end
+      # rubocop: disable Style/CombinableLoops
+      for i in self
+        return false if i == false
+      end
+      # rubocop: enable Style/CombinableLoops
+      return to_enum(:my_any?)
+    end
+    bul = false
+    arr = self
+    arr.my_each { |x| bul = true if yield(x) }
+    bul
+  end
+
+  ## my_all method
+  # rubocop: disable Metrics/CyclomaticComplexity
+  def my_all?
+    unless block_given?
+      for i in self
+        return false if i == false || i.nil?
+      end
+      # rubocop: disable Style/CombinableLoops
+      for i in self
+        return true if i == true
+      end
+      # rubocop: enable Style/CombinableLoops
+      return to_enum(:my_all?)
+    end
+    bul = false
+    arr = self
+    arr.my_each { |x| bul = true if yield(x) }
+    bul
+  end
+  # rubocop: enable Metrics/CyclomaticComplexity
+
+  ## my_none method
+  def my_none?
+    unless block_given?
+      for i in self
+        return false if i == true
+      end
+      # rubocop: disable Style/CombinableLoops
+      for i in self
+        return true if i == false
+      end
+      # rubocop: enable Style/CombinableLoops
+      return to_enum(:my_none?)
+    end
+    bul = true
+    arr = self
+    arr.my_each { |x| bul = false if yield(x) }
+    bul
+  end
+
+  ## my_count method
+  def my_count(args = nil)
+    counter = 0
+    unless args.nil?
+      for i in self
+        counter += 1 if i == args
+       end
+      return counter
+    end
+    counter
+  end
+
+  ## my_map method
+  def my_map(&factor)
+    return to_enum(:my_map) unless block_given?
+
+    arr = self
+    new_arr = []
+    if factor.nil?
+      arr.my_each { |x| new_arr << x if factor.call(x) }
+      arr
+    else
+      arr.my_each { |x| new_arr << x if yield(x) }
+      new_arr
+    end
+  end
+
+  # rubocop: disable Metrics/CyclomaticComplexity
+  # rubocop: disable Metrics/PerceivedComplexity
+  def my_inject(*args)
+    return min.length if args.length.zero? and self[0].is_a? String
+
+    if args.length == 1
+      if args.first.is_a?(Symbol)
+        sum = args.first
+      else
+        result = args.first
+      end
+    end
+    result = args.first, sum = args.last if args.length == 2
+    result ||= 0
+    my_each { |x| result = block_given? ? yield(result, x) : result.send(sum, x) }
+    result
+  end
```

Comment:
- [ ] #my_inject raises a "LocalJumpError" when no block or argument is given
      Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError).
`expected LocalJumpError, got #<TypeError: nil is not a symbol nor a string> with backtrace:`

- [ ] #my_inject searches for the longest word in an array of strings
      Failure/Error: expect(words.my_inject(&search)).to eq words.inject(&search)      
        expected: "blade"
             got: 5

- [ ] #my_inject when a block is given without an initial value combines all elements of enum by applying a binary operation, specified by a block::range
      Failure/Error: actual = range.my_inject { |prod, n| prod * n }
`undefined method `[]' for 5..50:Range`

- [ ] #my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method:range
      Failure/Error: expect(range.my_inject(:*)).to eq range.inject(:*)      
        expected: 1267253883404724085150525340252698701849068398706688000000000000
             got: 0